File: VB/DxBlazorApplication1/Models/UIModels/CustomAppointmentFormInfo.vb
Imports DevExpress.Blazor
Imports System.Diagnostics

Namespace DxBlazorApplication1.Models.UIModels

    Public Class CustomAppointmentFormInfo
        Inherits SchedulerAppointmentFormInfo

        Public Sub New(ByVal AppointmentItem As DxSchedulerAppointmentItem, ByVal DataStorage As DxSchedulerDataStorage, ByVal scheduler As DxScheduler)
            MyBase.New(AppointmentItem, DataStorage, scheduler)
        End Sub

        Public Property Grade As String?
            Get
                Return CustomFields("Grade")?.ToString()
            End Get

            Set(ByVal value As String?)
                CustomFields("Grade") = value
            End Set
        End Property
    End Class
End Namespace

File: CS/DxBlazorApplication1/Models/UIModels/CustomAppointmentFormInfo.cs
using DevExpress.Blazor;
using System.Diagnostics;

namespace DxBlazorApplication1.Models.UIModels
{
    public class CustomAppointmentFormInfo : SchedulerAppointmentFormInfo {
        public CustomAppointmentFormInfo(DxSchedulerAppointmentItem AppointmentItem,
                DxSchedulerDataStorage DataStorage, DxScheduler scheduler) : base(AppointmentItem, DataStorage, scheduler) { }

        public string? DegreeLevel {
            get { return CustomFields["DegreeLevel"]?.ToString(); }
            set { CustomFields["DegreeLevel"] = value; }
        }
    }
}
